src/Functions/array/arraySort.cpp

```diff
@@ -212,64 +212,211 @@ ColumnPtr ArraySortImpl<positive, is_partial>::execute(
-
-To retain only the sorted elements use `arrayResize`:
-[example:retain_sorted]
-
-If the `func` function is specified, sorting order is determined by the result of the `func`
-function applied to the elements of the array.
-[example:lambda_simple]
-
-If `func` accepts multiple arguments, the `arrayPartialSort` function is passed several arrays
-that the arguments of `func` will correspond to.
-[example:lambda_complex]
-
-For more details see documentation of `arraySort`.
-)",
-        .examples{
-            {"simple_int", "SELECT arrayPartialSort(2, [5, 9, 1, 3])", ""},
-            {"simple_string", "SELECT arrayPartialSort(2, ['expenses','lasso','embolism','gladly'])", ""},
-            {"retain_sorted", "SELECT arrayResize(arrayPartialSort(2, [5, 9, 1, 3]), 2)", ""},
-            {"lambda_simple", "SELECT arrayPartialSort((x) -> -x, 2, [5, 9, 1, 3])", ""},
-            {"lambda_complex", "SELECT arrayPartialSort((x, y) -> -y, 1, [0, 1, 2], [1, 2, 3]) as res", ""}},
-        .category = FunctionDocumentation::Category::Array});
-
-    factory.registerFunction<FunctionArrayPartialReverseSort>(FunctionDocumentation{
-        .description=R"(
-Returns an array of the same size as the original array where elements in range `[1..limit]`
-are sorted in descending order. Remaining elements `(limit..N]` shall contain elements in unspecified order.
-[example:simple_int]
-[example:simple_string]
-
-To retain only the sorted elements use `arrayResize`:
-[example:retain_sorted]
-
-If the `func` function is specified, sorting order is determined by the result of the `func`
-function applied to the elements of the array.
-[example:lambda_simple]
-
-If `func` accepts multiple arguments, the `arrayPartialSort` function is passed several arrays
-that the arguments of `func` will correspond to.
-[example:lambda_complex]
-
-For more details see documentation of `arraySort`.
-)",
-        .examples{
-            {"simple_int", "SELECT arrayPartialReverseSort(2, [5, 9, 1, 3])", ""},
-            {"simple_string", "SELECT arrayPartialReverseSort(2, ['expenses','lasso','embolism','gladly'])", ""},
-            {"retain_sorted", "SELECT arrayResize(arrayPartialReverseSort(2, [5, 9, 1, 3]), 2)", ""},
-            {"lambda_simple", "SELECT arrayPartialReverseSort((x) -> -x, 2, [5, 9, 1, 3])", ""},
-            {"lambda_complex", "SELECT arrayPartialReverseSort((x, y) -> -y, 1, [0, 1, 2], [1, 2, 3]) as res", ""}},
-        .category = FunctionDocumentation::Category::Array});
+    FunctionDocumentation::Description description = R"(
+Sorts the elements of the provided array in ascending order.
+If a lambda function `f` is specified, sorting order is determined by the result of
+the lambda applied each element of the array.
+If the lambda accepts multiple arguments, the `arraySort` function is passed several
+arrays that the arguments of `f` will correspond to.
+
+If the array to sort contains `-Inf`, `NULL`, `NaN`, or `Inf` they will be sorted in the following order:
```

Comment:
L. 224-227 is wrong, pls see https://fiddle.clickhouse.com/c24fa139-0d99-4ad3-8191-5b61b1161998